### PR TITLE
Enable WASM for linux and android.

### PR DESCRIPTION
**Pull request by ghost**
b/277955457

### CODE REVIEW

**Review by johnxwork — APPROVED** [review]
LGTM. It might be worthwhile to diff v8 with this change against the upstream V8 version to see if we missed anything.

**Comment by kaidokert** [comment]
I'll trust John as the expert for review here. But please add bug ID into commit message so lint / bug id check passes !

File: starboard/android/arm/args.gn [hunk 1]
```diff
@@ -15,3 +15,4 @@
 target_platform = "android-arm"
 target_os = "android"
 target_cpu = "arm"
+v8_enable_webassembly = true
```
Comment (y4vor) [inline starboard/android/arm/args.gn:18]: The flag should be turned on in starboard/linux/shared/platform_configuration/configuration.gni and andstarboard/android/shared/platform_configuration/configuration.gni

The args.gn is only used for local builds.

**Comment by kaidokert** [comment]
Last Win32 builds crashed - this may or may not be relevant. I'd recommend trying to re-run the failed jobs once to see if it's consistent.

File: third_party/v8/src/wasm/wasm-code-manager.cc [hunk 2]
```diff
@@ -178,14 +178,14 @@ std::unique_ptr<const byte[]> WasmCode::ConcatenateBytes(
   size_t total_size = 0;
   for (auto& vec : vectors) total_size += vec.size();
   // Use default-initialization (== no initialization).
-  byte* result = new byte[total_size];
```
Comment (y4vor) [inline third_party/v8/src/wasm/wasm-code-manager.cc:181]: If this is the original code, it had problems building on some of the 1P platforms.

**Review by y4vor — APPROVED** [review]

**Pull request merged**